### ISSUE TITLE
Naming Conventions

### ISSUE THREAD

**Issue by vsoch** [vsoch/freegenes#114]
As discussed in https://github.com/vsoch/freegenes/issues/113, since there are no naming conventions for various objects (aside from tags) we can't easily parse a search query to do something like remove invalid / uninteresting characters, as we might actually lose a result. This issue is opened for discussion and future work around developing a standard. What will need to be done is:

 - creating a standard for different types that can be searched
 - updating all models to conform to the standard
 - then doing a custom save to convert the input value to the standard, and if it doesn't pass validation (likely a regexp) then throw an error (return message to user)

Once this is done, the search query can be further parsed without worrying about losing content.

**Comment by marc4-stanford** [comment]
What is considered white space/word delmiters?  Are a space, comma, underscore, dash, period, end of line, etc all treated word terminators?

**Comment by vsoch** [comment]
This is largely up to the FreeGenes team to decide - with regular expressions we can parse the search query input however is desired. Currently there is no parsing.

**Comment by vsoch** [comment]
You can read about queries [here](https://docs.djangoproject.com/en/2.2/topics/db/queries/) we have a custom query based on the model type that looks over a subset of fields using icontains. For example:

```python
def distributions_query(q):
    '''specific search for distributions
    '''
    return Distribution.objects.filter(
                    Q(name__icontains=q) |
                    Q(description__icontains=q)).distinct()
```
where q is whatever is provided by the user, and icontains ensures that casing doesn't matter.  See [here](https://github.com/vsoch/freegenes/blob/master/fg/apps/base/views/search.py#L65) for the functions and full logic for each model.
